src/app/page.jsx

```diff
@@ -37,8 +37,12 @@ async function fetchVotableSupply() {
   return getVotableSupply();
 }
 
-export default async function Home() {
-  const proposals = await fetchProposals();
+export default async function Home({ searchParams }) {
+  const filter = searchParams?.filter
+    ? proposalsFilterOptions.everything.filter
+    : proposalsFilterOptions.relevant.filter;
+
+  const proposals = await fetchProposals(filter);
```

Comment:
@stepandel, I solved the refresh issue by adding useEffect to the ProposalList.jsx. This similar pattern was used on the voters' page. I generally try to avoid useEffect, but I think it works in this case. I did some QA on the test deployment and didn't encounter any issues, although it's hard to test when the query returns the same results for the first page. 

```
  useEffect(() => {
    setPages([initialProposals]);
    setMeta(initialProposals.meta);
  }, [initialProposals]);

```

Reply:
I'm noticing some weirdness as some proposals are not showing up when they supposed to.

Reply:
https://www.loom.com/share/be97ca0824834eba806da581aeec69d7?sid=fca6d309-3e6f-4a98-a6cb-af7863b15724

The behaviour has changed, but some proposals are disappearing